Automattic/wp-calypso
# Sites Dashboard: Add the primary action to open the site panel (Remove "click-on-row" behavior)

Related to https://github.com/Automattic/dotcom-forge/issues/9734
Also close https://github.com/Automattic/dotcom-forge/issues/9904

## Proposed Changes

This PR adds the primary action to open the site panel and remove the "click-on-row" behavior completely. 

- Removed the “row click” behavior from the table view
- Prepared the primary action to open the site panel
- Expanded the clickable area of the SiteField and moved the site URL and WP Admin links to actions. Also, added a hover color on the SiteField, but this will be revisited when we migrate Core's native primary/media field in https://github.com/Automattic/dotcom-forge/issues/9735.
- Added track events to know the trigger for the site panel opening. 

|view| before | after |
|--------|--------|--------|
|table view| <img width="1407" alt="Screenshot 2024-11-21 at 13 26 14" src="https://github.com/user-attachments/assets/dfe6fd27-00d9-4e68-a3a3-32ecbca8bdb2">the entire row is clickable | <img width="1307" alt="Screenshot 2024-11-21 at 13 25 42 copy 2" src="https://github.com/user-attachments/assets/f69038b1-15da-4b02-ac83-cecc575c6877"> |
|list view (when the panel is open)| <img width="407" alt="Screenshot 2024-11-21 at 13 26 59" src="https://github.com/user-attachments/assets/ace5e969-66e3-4673-92a1-9fb8e596992e">the entire row is clickable | <img width="403" alt="Screenshot 2024-11-21 at 13 27 09" src="https://github.com/user-attachments/assets/8cb9b83a-9aac-475c-88cf-cc2920299270">the entire row is clickable | 

Context: pfYzsZ-Mq-p2#comment-670

<details><summary>ver.1</summary>
<p>

<img width="1435" alt="Screenshot 2024-11-18 at 15 52 57" src="https://github.com/user-attachments/assets/a2bb2eca-4323-4ea1-a48d-32ff4ff74bf3">

</p>
</details> 


## Why are these changes being made?


Please refer to pbxlJb-6EH-p2#comment-4044 and https://github.com/Automattic/dotcom-forge/issues/9734 for more context. 

## Testing Instructions



* Go to /sites.
* Select any site to open the panel. You should click the ">" icon on the rightmost of the row. 
* Try to break this fly-out panel behavior.

## Pre-merge Checklist
